src/Acme.UI/Infrastructure/Routing/RouteParameters.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace Acme.UI.Infrastructure.Routing
+{
+    public class RouteParameters
+    {
+        public Type DataSourceType { get; private set; }
+
+        public string SearchNodeByName { get; private set; }
+        public string SearchNodeByNameFormatted { get; private set; }
+
+        public string RootNodeName { get; private set; }
+
+        public RouteParameters(Type dataSourceType, string searchNodeByName, string rootNodeName)
+        {
+            DataSourceType = dataSourceType;
+            SearchNodeByName = searchNodeByName;
+            SearchNodeByNameFormatted = searchNodeByName.Replace("-", " ");
```

Comment:
There is a built in function called ToUrlSegement
https://our.umbraco.com/documentation/reference/routing/request-pipeline/outbound-pipeline

I think you are taking care of dashes but what about other special characters like ampersand etc.

Reply:
Here the purpose is to convert the url to a name used for the node in the data hierarchy. I think "ToUrlSegment" doing the inverse of it.

Reply:
As per my investigation the generated Url for "acme\Test&^%*#-_=+$%@!star223" was "test-_-star223". umbraco trims almost all special characters (except hyphen and underscore). 

In SearchNodeByNameFormatted = searchNodeByName.Replace("-", " ");
we want to replace hyphen with space so that we could match it with athlete name. I don't think umbraco would be capable to change test-_-star223 to "Test&^%*#-_=+$%@!star223" in our case. let me know what you think.